tutor/src/components/task/progress/milestones.cjsx

```diff
@@ -27,32 +35,40 @@ Milestone = React.createClass
       keyEvent.preventDefault()
 
   render: ->
-    {goToStep, crumb, currentStep} = @props
+    {goToStep, crumb, currentStep, stepIndex} = @props
 
-    isCurrent = crumb.key is currentStep
+    isCurrent = stepIndex is currentStep
 
-    classes = classnames 'milestone',
+    classes = classnames 'milestone', "milestone-#{crumb.type}",
       'active': isCurrent
 
-    {title} = crumb.data
-    title ?= StepTitleStore.get(crumb.data.id)
+    previewText = StepTitleStore.get(crumb.id) if crumb.id?
+
+    switch crumb.type
+      when 'end'
+        previewText = "#{crumb.task.title} Completed"
+
+      when 'coach'
+        previewText = TITLES[SPACED_PRACTICE_GROUP]
+
+      when INTRO_ALIASES[SPACED_PRACTICE_GROUP]
+        previewText = TITLES[SPACED_PRACTICE_GROUP]
 
-    previewText = title
-    if crumb.type is 'end'
-      previewText = "#{previewText} Completed"
+      when INTRO_ALIASES[PERSONALIZED_GROUP]
+        previewText = TITLES[PERSONALIZED_GROUP]
 
-    if crumb.data.type is 'coach'
-      previewText = 'Spaced Practice'
+      when INTRO_ALIASES[TWO_STEP_ALIAS]
+        previewText = TITLES[TWO_STEP_ALIAS]
 
```

Comment:
Not really important, but you could also assign from the results of the `switch`, i.e.:
```
previewText = switch crumb.type
       when 'end' then "#{crumb.task.title} Completed"
...
```